components/RegistrationForm/RegistrationForm.js

```diff
@@ -0,0 +1,99 @@
+import React, { Component } from 'react';
+import { Formik, Field } from 'formik';
+import * as Yup from 'yup';
+import { minPasswordCharNum, validationErrorMessages } from 'common/constants/validations';
+import { isMinPasswordStrength } from 'common/utils/validator-utils';
+import Button from 'components/_common_/Button/Button';
+import Form from 'components/_common_/Form/Form';
+import Input from 'components/_common_/Form/Input/Input';
+
+const registrationSchema = Yup.object().shape({
+  email: Yup.string()
+    .required('Required')
+    .email(validationErrorMessages.email),
+  'confirm-email': Yup.string()
+    .required('Required')
+    .oneOf([Yup.ref('email')], validationErrorMessages.emailMatch),
+  password: Yup.string()
+    .required('Required')
+    .min(minPasswordCharNum, validationErrorMessages.length(minPasswordCharNum))
+    .test('password-strength', validationErrorMessages.password, isMinPasswordStrength),
+  'confirm-password': Yup.string()
+    .required('Required')
+    .oneOf([Yup.ref('password')], validationErrorMessages.passwordMatch),
+});
```

Comment:
`Yup` seems really nice, I have never heard of it before.

Reply:
Yeah we used it at AutoGravity. It's recommended in Formik's docs.